include/swift/AST/DiagnosticsParse.def

```diff
@@ -762,6 +762,13 @@ ERROR(extra_rbracket,PointsToFirstBadToken,
       "unexpected ']' in type; did you mean to write an array type?", ())
 ERROR(extra_colon,PointsToFirstBadToken,
       "unexpected ':' in type; did you mean to write a dictionary type?", ())
+WARNING(subscript_array_element, none,
+        "unexpected subscript in array literal; did you mean to write two "
+        "separate elements instead?", ())
+NOTE(subscript_array_element_fix_it_add_comma, none, "add a separator between "
+     "the elements", ())
+NOTE(subscript_array_element_fix_it_remove_space, none,
+     "remove the space between the elements to silence this warning", ())
```

Comment:
This gives the impression that the parser is already treating what's written as two elements or that the user intended to write two elements (since it's saying that there's space "between the elements"), when in fact the parser is already treating it as a subscript and the user intended to write a subscript if he or she applies this fix-it. I'd suggest:

"remove whitespace before subscript to silence this warning"